staging/src/k8s.io/apiserver/pkg/endpoints/handlers/apply.go

```diff
@@ -56,9 +56,11 @@ func (p *applyPatcher) extractLastIntent(obj runtime.Object, workflow string) (m
 		return nil, fmt.Errorf("couldn't get accessor: %v", err)
 	}
 	last := make(map[string]interface{})
-	if accessor.GetLastApplied()[workflow] != "" {
-		if err := json.Unmarshal([]byte(accessor.GetLastApplied()[workflow]), &last); err != nil {
-			return nil, fmt.Errorf("couldn't unmarshal last applied field: %v", err)
+	if _, ok := accessor.GetLastApplied()[workflow]; !ok {
+		// TODO: fix this
+		last, err = runtime.DefaultUnstructuredConverter.ToUnstructured(accessor.GetLastApplied()[workflow])
```

Comment:
Yeah, this doesn't necessarily need to be unstructured, just deserialized? Or does it?